### PR TITLE
Fix/1646 wcag page title

### PR DESCRIPTION
**Pull request by adamhaeger**
## Description

Implements better titles for our pages based on this: https://digdir.atlassian.net/wiki/spaces/BTA/pages/2815426563/URL+arkitektur+og+sidetitler

## Related Issue(s)

- closes https://github.com/Altinn/app-frontend-react/issues/1646

## Verification/QA

- Manual functionality testing
  - [x] I have tested these changes manually
  - [ ] Creator of the original issue (or service owner) has been contacted for manual testing (or will be contacted when released in alpha)
  - [ ] No testing done/necessary
- Automated tests
  - [ ] Unit test(s) have been added/updated
  - [x] Cypress E2E test(s) have been added/updated
  - [ ] No automatic tests are needed here (no functional changes/additions)
  - [ ] I want someone to help me make some tests
- UU/WCAG ([follow these guidelines](https://aksel.nav.no/god-praksis/artikler/utvikling?tema=universell-utforming) until we have our own)
  - [ ] I have tested with a screen reader/keyboard navigation/automated wcag validator
  - [ ] No testing done/necessary (no DOM/visual changes)
  - [ ] I want someone to help me perform accessibility testing
- User documentation @ [altinn-studio-docs](https://github.com/Altinn/altinn-studio-docs)
  - [ ] Has been added/updated
  
  - [ ] No functionality has been changed/added, so no documentation is needed
  - [ ] I will do that later/have created an issue
  
- Support in Altinn Studio
  - [ ] Issue(s) created for support in Studio
  
  - [ ] This change/feature does not require any changes to Altinn Studio
- Sprint board
  - [ ] The original issue (or this PR itself) has been added to the Team Apps project and to the current sprint board
  - [ ] I don't have permissions to do that, please help me out
- Labels
  - [ ] I have added a `kind/*` label to this PR for proper release notes grouping
  - [ ] I don't have permissions to add labels, please help me out

### CODE REVIEW

**Review by bjosttveit — COMMENTED** [review]
~~What does the title look like in PDF? I cannot see any specific case for that here. Does it still use `appName - appOwner` there?
https://github.com/Altinn/app-lib-dotnet/pull/792~~

Edit: We update the watermark instead

File: src/components/form/Form.test.tsx [hunk 1]
```diff
@@ -58,6 +60,7 @@ describe('Form', () => {
 
   it('should render components and groups', async () => {
     await render();
+    screen.debug();
```
Comment (bjosttveit) [inline src/components/form/Form.test.tsx:63]: Should this be removed?

File: src/components/form/Form.tsx [hunk 2]
```diff
@@ -94,6 +101,9 @@ export function FormPage({ currentPageId }: { currentPageId: string | undefined
 
   return (
     <>
+      <Helmet>
+        <title>{`${getPageTitle(appName, langAsString(currentPageId), appOwner)}`}</title>
```
Comment (bjosttveit) [inline src/components/form/Form.tsx:105]: As Ole Martin pointed out in slack, we should probably check that the currentPageId actually has a textResource associated with it before adding it. And if not, default to just `appName - appOwner` which is what we have today

**Review by olemartinorg — APPROVED** [review]
The texts do not seem to match exactly as in the referenced confluence page (i.e. `confirm.title` is `Se over svarene dine før du sender inn` not `Bekreft`), but then again the confluence page is lacking translations for the 3 languages we have to support by default.

Also, I'm still questioning if this is somewhat of a breaking change, as with this we'd start requiring translations for page names for apps - which is not something we've required before. 

PS: Make sure you follow the checklist! This PR is (at least) missing labels, which would make it fall into the 'uncategorized' list in the next release.

File: src/components/form/Form.test.tsx [hunk 3]
```diff
@@ -11,6 +11,8 @@ import { renderWithInstanceAndLayout } from 'src/test/renderWithProviders';
 import type { CompExternal, ILayout } from 'src/layout/layout';
 import type { CompSummaryExternal } from 'src/layout/Summary/config.generated';
 
+jest.mock('react-helmet-async');
+
```
[the remark this answers is not in the document]
Comment (adamhaeger) [reply src/components/form/Form.test.tsx:15]: This is just needed for the tests to run. I havent found a way to actually test that the titles are set properly with jest so the tests are in cypress for now.

File: src/features/instantiate/containers/PartySelection.tsx [hunk 4]
```diff
@@ -227,6 +233,9 @@ export const PartySelection = () => {
 
   return (
     <InstantiationContainer>
+      <Helmet>
+        <title>{`${getPageTitle(appName, langAsString('party_selection.header'), appOwner)}`}</title>
+      </Helmet>
```
[the remark this answers is not in the document]
Comment (adamhaeger) [reply src/features/instantiate/containers/PartySelection.tsx:238]: Well then we need to check what route we are on etc to the set correct title right? Adds a lot of steps to see whats going on. Ideally the helmet component should be able to live in the top route component which would be cleaner but we have checks for tasktype etc that forces us to render it further down. I suggest we keep it this way for now, but should be revisited when we do some refactoring.

**Pull request merged**